lib/build/filterChoicesUtils.js

```diff
@@ -0,0 +1,116 @@
+import { SUPPORTED_FRONTEND_FOR_CUSTOM_UI, SUPPORTED_RECIPE_FOR_CUSTOM_UI } from "./constants.js";
+import { UIBuildType } from "./types.js";
+const getIsValueSupported = (value, supportedValues) => {
+    return supportedValues.includes(value);
+};
+/**
+ * Frontend filter strategy based on the selected UI build type.
+ */
+const filterFrontendByUiType = {
+    filterChoices(choices, answers) {
+        /**
+         * For pre-built UI, all choices are supported which are not part of SUPPORTED_FRONTEND_FOR_CUSTOM_UI.
+         */
+        if (answers.ui === UIBuildType.PRE_BUILT) {
+            return choices.filter((choice) => !getIsValueSupported(choice.value, SUPPORTED_FRONTEND_FOR_CUSTOM_UI));
+        }
+        /**
+         * For custom UI, only a subset of frontend are supported.
+         */
+        return choices.filter((choice) => getIsValueSupported(choice.value, SUPPORTED_FRONTEND_FOR_CUSTOM_UI));
+    },
+    validateUserArguments(userArguments) {
+        /**
+         * If the user has not selected a frontend, then it is valid as the value may be chosen during the inquirer prompt.
+         * For pre-built UI, which is the default option all frontend which are not part of SUPPORTED_FRONTEND_FOR_CUSTOM_UI are supported.
+         */
+        if (
+            !userArguments.frontend ||
+            (userArguments.ui === UIBuildType.PRE_BUILT &&
+                !getIsValueSupported(userArguments.frontend, SUPPORTED_FRONTEND_FOR_CUSTOM_UI))
```

Comment:
shouldnt this be something like `getIsValueSupported(userArguments.frontend, SUPPORTED_FRONTEND_FOR_PRE_BUILT_UI)` instead of NOT custom ui?

Reply:
Deleted the constant file and created a local function which takes the UI type and returns the value. I have done this for both the frontend and the recipe selection. It improved the code readability.